autoload/db/adapter/sqlserver.vim

```diff
@@ -35,18 +35,34 @@ function! s:server(url) abort
         \ (has_key(a:url, 'port') ? ',' . a:url.port : '')
 endfunction
 
+function! s:get_value_from_url(url, param)
+    return get(a:url.params, a:param, get(a:url.params, toupper(a:param[0]) . a:param[1 : -1], '0'))
+endfunction
+
 function! s:boolean_param_flag(url, param, flag) abort
-  let value = get(a:url.params, a:param, get(a:url.params, toupper(a:param[0]) . a:param[1 : -1], '0'))
+  let value = s:get_value_from_url(a:url, a:param)
   return value =~# '^[1tTyY]' ? [a:flag] : []
 endfunction
 
+function! s:value_param_flag(url, param, flag) abort
+  let value = s:get_value_from_url(a:url, a:param)
+  return value == '0' ? [] : [a:flag, value]
+endfunction
+
 function! db#adapter#sqlserver#interactive(url) abort
   let url = db#url#parse(a:url)
-  return ['sqlcmd', '-S', s:server(url)] +
-        \ s:boolean_param_flag(url, 'encrypt', '-N') +
+  " Legacy parameters left just for backward compatibility with older sqlcmd: secure and
+  " trustServerCertificate
```

Comment:
They're not legacy, keep them as is, and fix your additions to match. See https://learn.microsoft.com/en-us/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver16#properties.

Reply:
ah now I understand the naming ( I thought that parameters were derived from sqlcmd --help ) - but at least for `secure` parameter which is supposed to be `encrypt` there was change
```
In version 11.2.0 and up, encrypt has been changed from boolean to string, allowing for TDS 8.0 support when the property is set to strict.
```
Do you want to keep `secure` a boolean parameter and `encrypt` as string parameter or there should be just one parameter and type of value will be detected ?

Reply:
Since `-N` doesn't accept a string, and thus there's no way for us to do anything with`encrypt=strict`, I think we should just leave it as is for now?

Reply:
That is actually one of the reasons why I started this pull request - I am using latest `sqlcmd`:
```bash
❯ sqlcmd --version
sqlcmd: 1.0.0
```
and I cannot use `-N` as boolean parameter because it was changed
```bash
❯ sqlcmd -N
sqlcmd: error: --encrypt-connection: expected string value but got "EOL" (<EOL>)
❯ sqlcmd -N X
sqlcmd: error: --encrypt-connection must be one of "default","false","true","disable" but got "X"
❯ sqlcmd -N disable
unable to open tcp connection with host 'localhost:1433': dial tcp [::1]:1433: connect: connection refused
```
The idea was that I would add parameter while keeping old one so that the adapter is not broken for people with older version of `sqlcmd`.

Reply:
I rebased the PR and one more thing to note is that `encrypt` and `trustServerCertificate` are properties of connection according to [specification](https://learn.microsoft.com/en-us/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver16#properties) but `format`, `column-separator` and `trim-spaces` are options of `sqlcmd` client.